src/Elders.Cronus/Multitenancy/MultitenancyExtensions.cs

```diff
@@ -1,40 +1,27 @@
-﻿using System;
-using System.Collections.Generic;
-using Elders.Cronus.Pipeline.Hosts;
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
 
 namespace Elders.Cronus.Multitenancy
 {
-    public static class MultitenancyExtensions
-    {
-        public static T WithTenants<T>(this T self, ITenantList tenants) where T : ICronusSettings
-        {
-            Func<ITenantList> combinedTenants = () => new ClientTenantsIncludingElders(tenants);
-            self.Container.RegisterSingleton(typeof(ITenantList), combinedTenants);
-
-            return self;
-        }
-
-        public static T WithNoTenants<T>(this T self) where T : ICronusSettings
-        {
-            Func<ITenantList> combinedTenants = () => new ClientTenantsIncludingElders();
-            self.Container.RegisterSingleton(typeof(ITenantList), combinedTenants);
-
-            return self;
-        }
-    }
-
     public class ClientTenantsIncludingElders : ITenantList
     {
         List<string> tenants;
 
-        public ClientTenantsIncludingElders(ITenantList clientTenants = null)
+        public ClientTenantsIncludingElders(IConfiguration configuration)
         {
-            if (ReferenceEquals(null, clientTenants))
-                tenants = new List<string>();
-            else
-                this.tenants = new List<string>(clientTenants.GetTenants());
-
-            this.tenants.Add(CronusAssembly.EldersTenant);
+            tenants = new List<string>();
+            tenants.Add(CronusAssembly.EldersTenant);
+
+            string tenantsFromConfiguration = configuration["cronus_tenants"];
```

Comment:
String based contracts smell bad.

Reply:
What are the alternatives?

Reply:
configuration.CronusTenants
configuration[CronusConstants.Tenants] at least